packages/apps/reputation-oracle/server/src/modules/qualification/qualification.service.spec.ts

```diff
@@ -1,348 +1,337 @@
-import { mockConfig } from '../../../test/constants';
+import { UserQualificationRepository } from './user-qualification.repository';
+
+const mockQualificationRepository = createMock<QualificationRepository>();
+const mockUserQualificationRepository =
+  createMock<UserQualificationRepository>();
+const mockUserRepository = createMock<UserRepository>();
 
 describe('QualificationService', () => {
-  let qualificationService: QualificationService;
-  let qualificationRepository: QualificationRepository;
-  let userRepository: UserRepository;
+  let service: QualificationService;
 
-  beforeEach(async () => {
-    const moduleRef = await Test.createTestingModule({
+  beforeAll(async () => {
+    const module = await Test.createTestingModule({
       providers: [
-        {
-          provide: ConfigService,
-          useValue: {
-            get: jest.fn((key: string) => mockConfig[key]),
-            getOrThrow: jest.fn((key: string) => {
-              if (!mockConfig[key]) {
-                throw new Error(`Configuration key "${key}" does not exist`);
-              }
-              return mockConfig[key];
-            }),
-          },
-        },
         QualificationService,
         {
           provide: QualificationRepository,
-          useValue: createMock<QualificationRepository>(),
+          useValue: mockQualificationRepository,
+        },
+        {
+          provide: UserQualificationRepository,
+          useValue: mockUserQualificationRepository,
+        },
+        {
+          provide: UserRepository,
+          useValue: mockUserRepository,
         },
-        { provide: UserRepository, useValue: createMock<UserRepository>() },
         ServerConfigService,
+        ConfigService,
       ],
     }).compile();
 
-    qualificationService =
-      moduleRef.get<QualificationService>(QualificationService);
-    qualificationRepository = moduleRef.get<QualificationRepository>(
-      QualificationRepository,
-    );
-    userRepository = moduleRef.get<UserRepository>(UserRepository);
+    service = module.get<QualificationService>(QualificationService);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
   });
 
   describe('createQualification', () => {
-    it('should create a new qualification', async () => {
-      const createQualificationDto: CreateQualificationDto = {
-        reference: 'ref1',
-        title: 'title1',
-        description: 'desc1',
-        expiresAt: '2025-12-31T00:00:00.000Z',
+    it.each([faker.date.future({ years: 1 }), undefined])(
+      'should create a new qualification',
+      async (expiresAt) => {
+        const newQualification = {
+          title: faker.string.alpha(),
+          description: faker.string.alpha(),
+          expiresAt,
+        };
+
+        mockQualificationRepository.createUnique.mockResolvedValueOnce({
+          reference: faker.string.uuid(),
+          ...newQualification,
+        } as QualificationEntity);
```

Comment:
```suggestion
        mockQualificationRepository.createUnique.mockImplementationOnce(
          async (e) => e,
        );
```
Then below you will be able to check the logic on assigning `uuid` to `reference`